### PR TITLE
enable compression by default

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/artery/compress/CompressionSettings.scala [hunk 1]
```diff
@@ -13,21 +13,18 @@ import scala.concurrent.duration._
...
 private[akka] class CompressionSettings(_config: Config) {
   val enabled = _config.getBoolean("enabled")
-  @inline private def globalEnabled = enabled
 
   val debug = _config.getBoolean("debug")
 
   object actorRefs {
     private val c = _config.getConfig("actor-refs")
 
-    val enabled = globalEnabled && c.getBoolean("enabled")
```
Comment (ktoso) [inline akka-remote/src/main/scala/akka/remote/artery/compress/CompressionSettings.scala]: remove the flag in some string configs as well, AFAIR MaxThroughputSpec also sets this to `on` explicitly

**Comment by ktoso** [comment]
LGTM, looking forward to this :)

File: akka-remote/src/main/scala/akka/remote/artery/compress/CompressionSettings.scala (hunk 1 above)
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/compress/CompressionSettings.scala]: good point, updated

File: akka-remote/src/test/scala/akka/remote/artery/HandshakeRetrySpec.scala [hunk 2]
```diff
@@ -20,11 +20,10 @@ object HandshakeRetrySpec {
      akka {
        actor.provider = remote
        remote.artery.enabled = on
-       remote.artery.compression.enabled = on
```
Comment (ktoso) [inline akka-remote/src/test/scala/akka/remote/artery/HandshakeRetrySpec.scala]: yup, LGTM

**Pull request merged**